semantic_router/route.py

```diff
@@ -116,16 +115,18 @@ def _parse_route_config(cls, config: str) -> str:
             raise ValueError("No <config></config> tags found in the output.")
 
     @classmethod
-    def _generate_dynamic_route(cls, llm: BaseLLM, function_schema: Dict[str, Any]):
+    def _generate_dynamic_route(cls, llm: BaseLLM, function_schemas: List[Dict[str, Any]], route_name: str):
         logger.info("Generating dynamic route...")
 
+        formatted_schemas = "\n".join([json.dumps(schema, indent=4) for schema in function_schemas])
         prompt = f"""
-        You are tasked to generate a JSON configuration based on the provided
-        function schema. Please follow the template below, no other tokens allowed:
+        You are tasked to generate a single JSON configuration for multiple function schemas. 
+        Each function schema should contribute five example utterances. 
+        Please follow the template below, no other tokens allowed:
 
         <config>
         {{
-            "name": "<function_name>",
+            "name": "{route_name}",
```

Comment:
Before it was function schema function name, now it is route name?

Is this a necessary change?

Reply:
Yes. Prevoiusly it was just setting the dynamic route name to the same name as the function attached to that route. Which was logical at the time.

Now we can have multiple functions in a route, and this doesn't make as much sense. Instead I made it so that the user has provide a name for the route.

Reply:
okay understood, why does each function schema need five example utterances? We only have one set of utterances per `Route` object?

Reply:
each function might have quite different wordings for the input utterances that trigger it. So, I thought we should include a few utterances for each function in the single dynamic route, as that way we cover a greater variety of utterances that might trigger each functions argument generation.